src/gossip/ping_pong.zig

```diff
@@ -146,11 +144,13 @@ pub const PingCache = struct {
     /// Records a `Pong` if corresponding `Ping` exists in `pending_cache`
     pub fn recevied_pong(self: *Self, pong: *const Pong, socket: SocketAddr, now: Instant) bool {
         var peer_and_addr = newPubkeyAndSocketAddr(pong.from, socket);
-        if (self.pending_cache.peek(pong.hash)) |val| {
-            if (val == peer_and_addr) {
-                self.pings.pop(&peer_and_addr);
-                self.pongs.put(peer_and_addr, now);
-                self.pending_cache.pop(&pong.hash);
+        if (self.pending_cache.peek(pong.hash)) |pubkey_and_addr| {
+            const pubkey: Pubkey = pubkey_and_addr[0];
+            const addr: SocketAddr = pubkey_and_addr[1];
+            if (pubkey.equals(&pong.from) and addr.eql(&socket)) {
```

Comment:
std.meta.eql

Reply:
- may lead to incorrect result (ie, if one pubkey has `cache_str = null` and another has `cache_str` set it will fail